src/solver/optimisation/opt_construction_variables_optimisees_quadratique.cpp

```diff
@@ -52,8 +52,135 @@ void OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeQuadratique(PROBLEME_H
     {
         CorrespondanceVarNativesVarOptim->NumeroDeVariableDeLInterconnexion[Interco]
           = NombreDeVariables;
-        ProblemeAResoudre->TypeDeVariable[NombreDeVariables] = VARIABLE_BORNEE_DES_DEUX_COTES;
+        ProblemeAResoudre->TypeDeVariable[NombreDeVariables] = VARIABLE_BORNEE_INFERIEUREMENT;
         NombreDeVariables++;
     }
     ProblemeAResoudre->NombreDeVariables = NombreDeVariables;
 }
+
+void constructVariableENS(PROBLEME_HEBDO* ProblemeHebdo, HOURLY_CSR_PROBLEM& hourlyCsrProblem)
+{
+    int NumberOfVariables = 0;
+    int hour = hourlyCsrProblem.hourInWeekTriggeredCsr;
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
+    ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+    CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim;
+    CorrespondanceVarNativesVarOptim = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour];
+
+    // variables: ENS of each area inside adq patch
+    logs.debug() << " ENS of each area inside adq patch: ";
+    for (int area = 0; area < ProblemeHebdo->NombreDePays; ++area)
+    {
+        // Only ENS for areas inside adq patch are considered as variables
+        if (ProblemeHebdo->adequacyPatchRuntimeData.areaMode[area]
+            == Data::AdequacyPatch::physicalAreaInsideAdqPatch)
+        {
+            CorrespondanceVarNativesVarOptim->NumeroDeVariableDefaillancePositive[area]
+              = NumberOfVariables;
+            ProblemeAResoudre->TypeDeVariable[NumberOfVariables] = VARIABLE_BORNEE_DES_DEUX_COTES;
+            logs.debug() << NumberOfVariables << " ENS[" << area << "].-["
+                         << ProblemeHebdo->NomsDesPays[area] << "].";
+
+            NumberOfVariables++;
+        }
+    }
+    ProblemeAResoudre->NombreDeVariables = NumberOfVariables;
+}
+
+void constructVariableSpilledEnergy(PROBLEME_HEBDO* ProblemeHebdo,
+                                    HOURLY_CSR_PROBLEM& hourlyCsrProblem)
+{
+    int hour = hourlyCsrProblem.hourInWeekTriggeredCsr;
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
+    ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+    CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim;
+    CorrespondanceVarNativesVarOptim = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour];
+    int NumberOfVariables = ProblemeAResoudre->NombreDeVariables;
```

Comment:
```suggestion
    int& NumberOfVariables = ProblemeAResoudre->NombreDeVariables;
```
Then there is no need to copy it back to `ProblemeAResoudre` at the end.